documentation/configuration.md

```diff
@@ -842,6 +842,75 @@ In addition to enabling custom providers, `dotnet monitor` also allows you to di
 
 The Queue Message's payload will be the blob name (`<BlobPrefix>/<ArtifactName>`; using the above example with an artifact named `mydump.dmp`, this would be `artifacts/mydump.dmp`) that is being egressed to blob storage. This is designed to be easily integrated into an Azure Function that triggers whenever a new message is added to the queue, providing you with the contents of the artifact as a stream. See [Azure Blob storage input binding for Azure Functions](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-input?tabs=csharp#example) for an example.
 
+### S3 storage egress provider
+
+| Name | Type | Required | Description |
+|---|---|---|---|
+| endpoint | string | true | The URI of the S3 storage service |
+| bucketName | string | true | The name of the s3 Bucket to which the blob will be egressed |
+| userName | string | true | The user credential for accessing the s3 service |
+| secretsFile | string | false | Path to a file on disk which holds the user's password for accessing the s3 storage. If not provided the `password` property must be set. |
+| password | string | false | The user's password for accessing the s3 storage. If not provided the `SecretsFile` must be specified. |
+| copyBufferSize | string | false | The buffer size to use when copying data from the original artifact to the blob stream.|
+
+### Example S3 storage provider
+
+<details>
+  <summary>JSON with secret file</summary>
+
+  ```json
+  {
+      "Egress": {
+          "S3Storage": {
+              "monitorS3Blob": {
+                  "endpoint": "http://localhost:9000",
```

Comment:
Well, I guess you are right in case we are talking about Amazon S3. For alternative S3 implementations like minio you need to know the endpoint. I will adapt it to make it optional.